examples/api-routes-apollo-server-and-client/lib/init-apollo.js

```diff
@@ -0,0 +1,56 @@
+import ApolloClient from 'apollo-client'
+import { SchemaLink } from 'apollo-link-schema'
+import { InMemoryCache } from 'apollo-cache-inmemory'
+import { HttpLink } from 'apollo-link-http'
+import { schema } from '../apollo/schema'
+import gql from 'graphql-tag'
+
+let apolloClient = null
+
+function create (initialState) {
+  // Check out https://github.com/zeit/next.js/pull/4611 if you want to use the AWSAppSyncClient
+  const isBrowser = typeof window !== 'undefined'
+
+  const link = isBrowser
+    ? new HttpLink({
+      uri: '/api/graphql',
+      credentials: 'same-origin'
+    })
+    : new SchemaLink({ schema })
```

Comment:
Awesome addition!  Maybe we can remove `apollo-link-schema` from the client bundle by using an inline require or import.